File: src/main/java/dev/kensa/state/TestInvocationContext.java
package dev.kensa.state;

import java.lang.reflect.Method;

public class TestInvocationContext {
    private final Object testInstance;
    private final Method testMethod;
    private final Object[] testParameters;

    public TestInvocationContext(Object testInstance, Method testMethod, Object[] testParameters) {
        this.testInstance = testInstance;
        this.testMethod = testMethod;
        this.testParameters = testParameters;
    }

    public Object testInstance() {
        return testInstance;
    }

    public Method testMethod() {
        return testMethod;
    }

    public Object[] testParameters() {
        return testParameters;
    }
}

File: core/src/main/kotlin/dev/kensa/state/TestInvocationContext.kt
package dev.kensa.state

import dev.kensa.FixturesAndOutputs
import dev.kensa.fixture.Fixtures
import dev.kensa.outputs.CapturedOutputs
import dev.kensa.util.isKotlinClass
import java.lang.reflect.Method

class TestInvocationContext(val instance: Any, val method: Method, val arguments: Array<Any?>, val displayName: String, val startTimeMs: Long, val fixtures: Fixtures, val capturedOutputs: CapturedOutputs) {
    val fixturesAndOutputs = FixturesAndOutputs(fixtures, capturedOutputs)
    val isKotlin = instance::class.isKotlinClass
}